python/dgl/distributed/dist_graph.py

```diff
@@ -116,22 +118,50 @@ def _get_graph_from_shared_mem(graph_name):
 
     g = DGLGraph(gidx)
     g.ndata['local_node'] = _get_shared_mem_ndata(g, graph_name, 'local_node')
+    g.edata['local_edge'] = _get_shared_mem_edata(g, graph_name, 'local_edge')
     g.ndata[NID] = _get_shared_mem_ndata(g, graph_name, NID)
     g.edata[EID] = _get_shared_mem_edata(g, graph_name, EID)
     return g
 
+def _move_metadata_to_shared_mam(graph_name, num_nodes, num_edges, part_id,
+                                 num_partitions, node_map, edge_map):
+    ''' Move all metadata to the shared memory.
+
+    We need these metadata to construct graph partition book.
+    '''
+    meta = _move_data_to_shared_mem_array(F.tensor([num_nodes, num_edges,
+                                                    num_partitions, part_id]),
+                                          _get_ndata_path(graph_name, 'meta'))
+    node_map = _move_data_to_shared_mem_array(node_map, _get_ndata_path(graph_name, 'node_map'))
+    edge_map = _move_data_to_shared_mem_array(edge_map, _get_edata_path(graph_name, 'edge_map'))
+    return meta, node_map, edge_map
+
```

Comment:
I think the _move_metadata_to_shared_mem() will be changed by our new rpc code. All the information shared by client and server will be put into ServerState object.

Reply:
i know that. btw, why is ServerState in the RPC code. RPC should just be used for communication.

Reply:
@zheng-da Agree. We can adjust that later.